src/java/ognl/Ognl.java

```diff
@@ -90,6 +91,23 @@
 public abstract class Ognl
 {
 
+    private static Integer expressionMaxLength;
```

Comment:
You might want to declare it `volatile`, to ensure visibility of any change to all threads:
` private static volatile Integer expressionMaxLength;`